### PR TITLE
hyprctl: add decorations

### PR DESCRIPTION
**Pull request by MightyPlaza**
adds `hyprctl decorations`
allows user to see priority for each decoration so they can (in the future) directly set priority for each one

https://github.com/hyprwm/hyprland-wiki/pull/427

### CODE REVIEW

**Comment by Eclextic** [comment]
Is this for plugin use?
If so that could be quite useful...

File: hyprctl/main.cpp [hunk 1]
```diff
@@ -423,6 +424,8 @@ int main(int argc, char** argv) {
         request(fullRequest, 1);
     else if (fullRequest.contains("/keyword"))
         request(fullRequest, 2);
+    else if (fullRequest.contains("/decoration"))
```
Comment (vaxerski) [inline hyprctl/main.cpp:427]: mismatch: help says decorations, here it's decoration? Later in hyprland it's decorations again?

File: src/render/decorations/IHyprWindowDecoration.cpp [hunk 2]
```diff
@@ -26,4 +26,8 @@ eDecorationLayer IHyprWindowDecoration::getDecorationLayer() {
 
 uint64_t IHyprWindowDecoration::getDecorationFlags() {
     return 0;
-}
\ No newline at end of file
+}
+
+std::string IHyprWindowDecoration::getDisplayName() {
+    return std::string("Generic Decoration");
```
Comment (vaxerski) [inline src/render/decorations/IHyprWindowDecoration.cpp:32]: Unknown would be better. Also no need for `std::string()`

File: hyprctl/main.cpp (hunk 1 above)
Comment (MightyPlaza) [reply hyprctl/main.cpp]: done

File: src/render/decorations/IHyprWindowDecoration.cpp (hunk 2 above)
Comment (MightyPlaza) [reply src/render/decorations/IHyprWindowDecoration.cpp]: done

**Comment by MightyPlaza** [comment]
> Is this for plugin use? If so that could be quite useful...

anything that uses that uses the `IHyprWindowDecoration` class

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**